pkg/datapath/iptables/iptables.go

```diff
@@ -689,52 +687,135 @@ func (m *IptablesManager) installStaticProxyRules() error {
 		}
 		if err == nil && m.haveSocketMatch {
 			// Direct inbound TPROXYed traffic towards the socket
-			err = runProg("ip6tables", m.inboundProxyRedirectRule("-A"), false)
+			err = ip6tables.runProg(m.inboundProxyRedirectRule("-A"), false)
 		}
 	}
 	return err
 }
 
-// install or remove rules for a single proxy port
-func (m *IptablesManager) iptProxyRules(cmd string, proxyPort uint16, ingress bool, name string) error {
-	// Redirect packets to the host proxy via TPROXY, as directed by the Cilium
-	// datapath bpf programs via skb marks (egress) or DSCP (ingress).
+func (m *IptablesManager) doCopyProxyRules(prog iptablesInterface, table string, re *regexp.Regexp, match, oldChain, newChain string) {
+	output, err := prog.runProgCombinedOutput(append(m.waitArgs, "-t", table, "-S"), true)
+	if err != nil {
+		log.WithFields(logrus.Fields{
+			"table": table,
+			"prog":  prog.getProg(),
+		}).WithError(err).Warning("Cannot list rules in table. Layer 7 proxy port may get reallocated, which could cause disruption for traffic selected by L7 policy", prog.getProg(), table)
+		return
+	}
+	scanner := bufio.NewScanner(bytes.NewReader(output))
+	for scanner.Scan() {
+		rule := scanner.Text()
+		if re.MatchString(rule) && strings.Contains(rule, match) {
+			log.WithField(logfields.Object, logfields.Repr(rule)).Debugf("Considering copying %s TPROXY rule from %s to %s", prog, oldChain, newChain)
+			args, err := shellwords.Parse(strings.Replace(rule, oldChain, newChain, 1))
+			if err != nil {
+				log.WithFields(logrus.Fields{
+					"table": table,
+					"prog":  prog.getProg(),
+				}).WithError(err).WithField(logfields.Object, rule).Warnf("Unable to parse %s TPROXY rule, disruption to traffic seleccted by L7 policy possible", prog)
```

Comment:
nit, if `prog` is part of the structured logging, then I don't think it needs to be part of the message any more. In general I encourage avoiding the Warnf() variations on logging functions but in practice if the variable content is only a couple of variations it's not as bad as if it can be any random dynamic content. We also can combine the `WithFields()` and `WithField()`. There's also a typo here.

```suggestion
				log.WithFields(logrus.Fields{
					"table": table,
					"prog":  prog.getProg(),
					logfields.Object: rule,
				}).WithError(err).Warn("Unable to parse TPROXY rule, disruption to traffic selected by L7 policy possible")
```

(Similar feedback for other log statements in the PR).

Reply:
I'm convinced by now that my keyboard is acting up, doubling letters randomly...
Fixed :-)

Reply:
Not gonna point my fingers at any particular fruit-branded manufacturers... ;)